DigitalEcosystem/publication/bandgap2DMaterials/2dm_bandgap.py

```diff
@@ -0,0 +1,588 @@
+
+# In[]:
+
+
+roost_train_results = pd.read_csv("roost/roost_train_predictions.csv", index_col="material_id")
+roost_test_results  = pd.read_csv("roost/roost_test_predictions.csv", index_col="material_id")
+
+
+# In[]:
+
+
+DigitalEcosystem.utils.figures.publication_parity_plot(train_y_true = roost_train_results['bandgap_target'],
+                                                       train_y_pred =  roost_train_results['bandgap_pred_n0'],
+                                                       test_y_true = roost_test_results['bandgap_target'],
+                                                       test_y_pred = roost_test_results['bandgap_pred_n0'],
+                                                       axis_label = "Bandgap (eV)",
+                                                       filename = "roost_2dm_bandgap_parity.jpeg")
+
+
+# In[]:
+
+
+print("Test Set Error Metrics")
+for key, fun in metrics.items():
+    value = fun(y_true=roost_test_results['bandgap_target'], y_pred=roost_test_results['bandgap_pred_n0'])
+    print(key,np.round(value,4))
+    
+print("\nTraining Set Error Metrics")
+for key, fun in metrics.items():
+    value = fun(y_true=roost_train_results['bandgap_target'], y_pred=roost_train_results['bandgap_pred_n0'])
+    print(key,np.round(value,4))
+
+
+# # SISSO
+# 
+# SISSO is a symbolic regression technique focused on creating interpretable machine learning models. 
+# 
+# Due to the exponential computational cost of running a SISSO model as the number of features and rungs increases, we need to restrict the feature space. To do that, we'll use LASSO-based feature selection (essentially we can look at how quickly LASSO extinguishes a variable to get an idea of its importance). 
+
+# In[]:
+
+
+sisso_feature_selector = sklearn.feature_selection.SelectFromModel(sklearn.linear_model.LassoCV(random_state=RANDOM_SEED),
+                                                                   threshold=-np.inf,
+                                                                   max_features=16,
+                                                                   prefit=False)
+sisso_feature_selector.fit(train_x, train_y.ravel())
+
+sisso_features = [col for (col, is_selected) in zip(train[descriptors].columns, sisso_feature_selector.get_support()) if is_selected]
+print("\n".join(sisso_features))
+
+
+# In[]:
+
+
+sisso_dir = "./sisso"
+os.makedirs(sisso_dir, exist_ok=True)
+
+sisso_data_train = train[target_column + sisso_features]
+sisso_data_test = test[target_column + sisso_features]
+
+sisso_data_train.to_csv(os.path.join(sisso_dir, 'sisso_train.csv'), index_label='2dm_id (unitless)')
+sisso_data_test.to_csv(os.path.join(sisso_dir, 'sisso_test.csv'), index_label='2dm_id (unitless)')
+
+
+# At this point, a SISSO model was run.
+
+# In[]:
+
+
+sisso_models = {
+    'r1_1term': lambda df: 5.908677334854443e+00 + \
+                           -4.182238907095930e-01 * (np.cbrt(df['ave:boiling_point'])),
```

Comment:
How did we get these "magical" numbers from the SISSO log? Please explain which log these are from in a comment.

Reply:
Added a note above the cell explaining where they're from